xskillscore/tests/test_comparative.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+import pytest
+
```

Comment:
I would add a test that checks all custom metrics  and runs through:

```python
        if observation is not None:
            # Compare the forecasts and observation using metric
            metric_f1o = metric(observation, forecast1, dim=[])
            metric_f2o = metric(observation, forecast2, dim=[])
```

Since right now this would break with any correlation metrics.

Reply:
Correlation metrics also should break.

Reply:
Based on our discussion above, agreed. Maybe explicitly test that correlation metrics break.

Reply:
You can run the sign test with correlation, but only if the correlation is evaluated along a dimension ( or dimensions) other than `time_dim`. The basic requirement of the sign test is that one can perform a comparison at each time step. So, for example, comparing the spatial correlation through time is valid.

I think we should enforce that `time_dim` is not in `dim` (see my above comment), but then there should be no issue with users wanting to use correlation as a metric.

Regarding this specific point, I think it's a good idea to add a test for all metrics. But, as @aaronspring points out, correlation should break with `dim=[]` because one can't compute correlation at a point.